go/arrow/compute/utils.go

```diff
@@ -81,3 +84,159 @@ func (b *bufferWriteSeeker) Seek(offset int64, whence int) (int64, error) {
 	b.pos = newpos
 	return int64(newpos), nil
 }
+
+func ensureDictionaryDecoded(vals ...arrow.DataType) {
+	for i, v := range vals {
+		if v.ID() == arrow.DICTIONARY {
+			vals[i] = v.(*arrow.DictionaryType).ValueType
```

Comment:
For the record, where does the actual dictionary decoding happen? (add a comment?)

Reply:
currently it's not implemented and will return an `ErrNotImplemented` before it gets here, but when I implement it i'll update this with a comment.

Reply:
comment added